app/helpers/concerns/foreman_teamdynamix/hosts_helper_extensions.rb

```diff
@@ -7,19 +7,47 @@ def teamdynamix_title
       SETTINGS[:teamdynamix][:title] || 'Team Dynamix'
     end
 
-    def teamdynamix_fields(host)
+    def teamdynamix_fields
       td_pane_fields = SETTINGS[:teamdynamix][:fields] || DEFAULT_TD_PANE_FIELDS
-      return [[_('Asset'), 'None Associated']] unless host.teamdynamix_asset_id
-      @asset = host.td_api.get_asset(host.teamdynamix_asset_id)
+      return [[_('Asset'), 'None Associated']] unless @host.teamdynamix_asset_id
+
+      get_teamdynamix_asset(@host.teamdynamix_asset_id)
+
+      # always display a link to the asset
+      fields = [ get_asset_uri ]
 
-      fields = []
       td_pane_fields.each do |field_name, asset_attr|
-        asset_attr_val = @asset[asset_attr]
-        fields += [[_(field_name.to_s), asset_attr_val]] if asset_attr_val.present?
+        asset_attr_val = @asset.has_key?(asset_attr) ? @asset[asset_attr] : get_nested_attrib_val(asset_attr)
+        fields += [[_(field_name.to_s), asset_attr_val]]
       end
       fields
     rescue StandardError => e
       [[_('Error'), e.message]]
     end
+
+    private
+    def get_asset_uri
+      api_url = SETTINGS[:teamdynamix][:api][:url]
+      asset_uri = api_url.split('api').first + @asset['Uri']
+      actual_link = "<a href='#{asset_uri}' target='_blank'>#{@asset['Uri']}</a>"
+      ['URI', actual_link.html_safe]
```

Comment:
You need to format the response with the _() method. This does language translations etc..
https://github.com/theforeman/foreman/blob/develop/app/helpers/hosts_helper.rb#L268

Reply:
ok